File: spring-webflux/webflux/websocket-coroutine/src/main/java/com/example/webflux/websocket/config/MappingConfig.java
package com.example.webflux.websocket.config;

import com.example.webflux.websocket.handler.ChatCoroutineWebSocketHandler;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping;

import java.util.Map;

@Configuration
public class MappingConfig {

    @Bean
    public SimpleUrlHandlerMapping simpleUrlHandlerMapping(
            ChatCoroutineWebSocketHandler chatWebSocketHandler
    ) {
        Map<String, ChatCoroutineWebSocketHandler> urlMapper = Map.of(
                "/chat", chatWebSocketHandler
        );

        SimpleUrlHandlerMapping mapping = new SimpleUrlHandlerMapping();
        mapping.setOrder(1);
        mapping.setUrlMap(urlMapper);

        return mapping;
    }
}

File: spring-webflux/webflux/websocket-coroutine/src/main/kotlin/com/example/webflux/websocket/config/MappingConfig.kt
package com.example.webflux.websocket.config

import com.example.webflux.websocket.handler.ChatCoroutineWebSocketHandler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping
import java.util.Map

@Configuration
class MappingConfig {

    @Bean
    fun simpleUrlHandlerMapping(
        chatWebSocketHandler: ChatCoroutineWebSocketHandler?
    ): SimpleUrlHandlerMapping {
        val urlMapper = mapOf(
            "/chat" to chatWebSocketHandler
        )

        return SimpleUrlHandlerMapping().also {
            it.order = 1
            it.urlMap = urlMapper
        }
    }
}